pkg/azure/blob/credentials.go

```diff
@@ -43,6 +72,125 @@ func GetCredentials(cfg azureconfig.Config, l hclog.Logger) (CredentialSet, erro
+		return CredentialSet{}, false, nil
+	}
+
+	if cfg.StorageAccount == "" {
+		return CredentialSet{}, true, errors.New("account is not set - cannot login with Azure CLI")
+	}
+
+	if cfg.StorageAccountResourceGroup == "" {
+		return CredentialSet{}, true, errors.New("resource-group is not set - cannot login with Azure CLI")
+	}
+
+	authorizer, err := auth.NewAuthorizerFromCLI()
+	if err != nil {
+		return CredentialSet{}, true, errors.Wrap(err, "Failed to login with Azure CLI")
+	}
+	subscriptionId := cfg.StorageAccountSubscriptionId
+	if subscriptionId == "" {
+		subscriptionId, err = getCurrentAzureSubscriptionFromCli()
+		if err != nil {
+			return CredentialSet{}, true, err
+		}
+	}
+	accountsClient := storage.NewAccountsClient(subscriptionId)
+	accountsClient.Authorizer = authorizer
+	err = accountsClient.AddToUserAgent(UserAgent)
+	if err != nil {
+		l.Debug(fmt.Sprintf("Error updating User Agent string for Azure: %v", err))
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	result, err := accountsClient.ListKeys(ctx, cfg.StorageAccountResourceGroup, cfg.StorageAccount, "")
+	if err != nil {
+		return CredentialSet{}, true, errors.Wrap(err, "Failed to get storage account keys")
+	}
+	storageAccountKey := (*result.Keys)[0]
+	cred, err := azblob.NewSharedKeyCredential(cfg.StorageAccount, *storageAccountKey.Value)
+	if err != nil {
+		return CredentialSet{}, true, errors.Wrap(err, "Failed to create storage account credential")
+	}
+	pipe := azblob.NewPipeline(cred, azblob.PipelineOptions{})
+	return CredentialSet{Credential: *cred, Pipeline: pipe}, true, nil
+
+}
+
+func getCurrentAzureSubscriptionFromCli() (string, error) {
+	var subscription AvailableSubscription
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", errors.Wrap(err, "Error getting home directory")
+	}
+	file, err := os.Open(path.Join(home, AzureDirectory, AzureProile))
+	if err != nil {
+		return "", errors.Wrap(err, "Error getting azure profile")
+	}
+	defer file.Close()
+
+	// azureProfile can have BOM and embedded BOM so use decoder and check for BOM rather than unmarshall
+
+	reader := bufio.NewReader(file)
+	if err := removeBOM(reader); err != nil {
+		return "", err
+	}
+
+	decoder := json.NewDecoder(reader)
+	if _, err := decoder.Token(); err != nil {
+		return "", errors.Wrap(err, "Error decoding opening json token")
+	}
+
+	property, err := decoder.Token()
+	if err != nil {
+		return "", errors.Wrap(err, "Error decoding subscriptions token")
+	}
+	if val, ok := property.(string); !ok || !strings.EqualFold(val, "subscriptions") {
+		return "", errors.Wrap(err, "Error gettting subscriptions property")
+	}
+
+	delim, err := decoder.Token()
+	if err != nil {
+		return "", errors.Wrap(err, "Error decoding json array delimiter")
+	}
+	if _, ok := delim.(json.Delim); !ok {
+		return "", errors.Wrap(err, "Error getting array delimiter")
+	}
+
+	for decoder.More() {
+
+		// azureProfile can have embedded BOM
```

Comment:
I am not seeing this happen when I login. There is just the BOM at the beginning, not nested in the document. Can you provide more information on when that happens, and make sure to include a test case that covers that scenario as well if it really does happen?

Reply:
I cant repro the embedded case in my tests either so I removed it (thought it was strange at the time), as I don't have that any longer I reverted to using unMarshall after checking for BOM and added a couple of tests for scenarios where BOM is present and is not present